vendor/models/offer.py

```diff
@@ -155,3 +156,56 @@ def get_best_currency(self, currency=DEFAULT_CURRENCY):
             return currency
 
         return DEFAULT_CURRENCY
+
+    def get_trial_amount(self):
+        return self.term_details.get('trial_amount', 0)
+
+    def get_trial_savings(self):
+        """
+        Returns the trial_amount savings if the offer has any trial occurrences.
+        """
+        if not self.has_trial_occurrences():
+            return 0
+
+        trial_savings = self.current_price() - self.term_details.get('trial_amount', 0)
+
+        if trial_savings < 0:
+            return 0
+
+        return trial_savings
+
+    def get_trial_discount(self):
+        """
+        Returns the trial_amount savings if the offer has any trial occurrences.
```

Comment:
what is this function used for? the description looks like a repeat of get_trial_savings

Reply:
Updated comments that make more sense.